### PR TITLE
Add uber-specific version of puppet-lint

### PR DESCRIPTION
**Pull request by wybczu**

### CODE REVIEW

File: src/lint/linter/UberArcanistPuppetLintLinter.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * A linter for Puppet files.
+ */
+final class ArcanistUberPuppetLintLinter extends ArcanistExternalLinter {
```
Comment (Pawka) [inline src/lint/linter/UberArcanistPuppetLintLinter.php]: Rename class name to match filename.

File: src/lint/linter/UberArcanistPuppetLintLinter.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  public function getLinterName() {
+    return 'UBERPUPPETLINT';
+  }
+
+  public function getLinterConfigurationName() {
+    return 'uber-puppet-lint';
+  }
+
+  public function getDefaultBinary() {
+    return 'scripts/uber-puppet-lint';
```
Comment (Pawka) [inline src/lint/linter/UberArcanistPuppetLintLinter.php]: Prefix with project's root directory.

File: src/lint/linter/UberArcanistPuppetLintLinter.php (hunk 1 above)
Comment (wybczu) [inline src/lint/linter/UberArcanistPuppetLintLinter.php]: Done.

File: src/lint/linter/UberArcanistPuppetLintLinter.php (hunk 2 above)
Comment (wybczu) [inline src/lint/linter/UberArcanistPuppetLintLinter.php]: Done.

**Comment by Pawka** [comment]
Looks good!

**Pull request merged**